storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -269,7 +276,10 @@ private void updateChainHead(
 
         final ChainHead previousChainHead = originalHead.get();
 
-        final UInt64 commonAncestorSlot = previousChainHead.findCommonAncestor(newChainHead);
+        final UInt64 commonAncestorSlot =
+            previousChainHead.findCommonAncestor(
+                newChainHead,
+                specProvider.atSlot(newChainHead.getSlot()).getConstants().getSlotsPerEpoch());
```

Comment:
Hmmm - this looks like the wrong constant.  We should add tests around this.  This error maybe suggests we should inject the full `specProvider` in cases like this - less likely to make this kind of mistake.  And this will let us dedupe the logic to extract the needed value (only happens in one place within the method call). 
```suggestion
                specProvider.atSlot(newChainHead.getSlot()).getConstants().getSlotsPerHistoricalRoot());
```

Reply:
i thought i checked everywhere :/ so many changes